src/go/intelligence/pipeline/fetch_alert.go

```diff
@@ -0,0 +1,100 @@
+package pipeline
+
+import (
+	"encoding/json"
+	"errors"
+	"go/intelligence/models"
+	"go/intelligence/utils"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/http/httputil"
+	"time"
+)
+
+// Module     : intelligence
+// Author     : Rahul Indra <indrarahul2013 AT gmail dot com>
+// Created    : Wed, 1 July 2020 11:04:01 GMT
+// Description: CERN MONIT infrastructure Intelligence Module
+
+//FetchAlert function for fetching all active alerts from AlertManager
+func FetchAlert() <-chan models.AmJSON {
+	fetchedData := make(chan models.AmJSON)
+	go func() {
+		data, err := get()
+		if err != nil {
+			log.Fatalf("Could not fetch alerts from AlertManager, error:%v\n", err)
+		}
+		for _, each := range data.Data {
+			fetchedData <- each
+		}
+		close(fetchedData)
+	}()
+	return fetchedData
+}
+
+//get function for get request on /api/v1/alerts alertmanager endpoint for fetching alerts.
+func get() (models.AmData, error) {
+
+	var data models.AmData
+	apiurl := utils.ValidateURL(utils.ConfigJSON.CMSMONURL, utils.ConfigJSON.GetAlertsAPI) //GET API for fetching all AM alerts.
+
+	req, err := http.NewRequest("GET", apiurl, nil)
+	if err != nil {
+		log.Printf("Request Error, error: %v\n", err)
+		return data, err
+	}
+	req.Header.Add("Accept-Encoding", "identity")
+	req.Header.Add("Accept", "application/json")
+
+	timeout := time.Duration(utils.ConfigJSON.HTTPTimeout) * time.Second
+	client := &http.Client{Timeout: timeout}
+
+	if utils.ConfigJSON.Verbose > 1 {
+		dump, err := httputil.DumpRequestOut(req, true)
+		if err == nil {
+			log.Println("Request: ", string(dump))
+		} else {
+			return data, err
+		}
+	}
+
+	resp, err := client.Do(req)
+	if err != nil {
+		log.Printf("Response Error, error: %v\n", err)
+		return data, err
+	}
+	if resp.StatusCode != http.StatusOK {
+		log.Printf("Http Response Code Error, status code: %d", resp.StatusCode)
+		return data, errors.New("Respose Error")
+	}
+
+	defer resp.Body.Close()
+
+	byteValue, err := ioutil.ReadAll(resp.Body)
+
+	if err != nil {
+		log.Printf("Unable to read JSON Data from AlertManager GET API, error: %v\n", err)
+		return data, err
+	}
+
+	err = json.Unmarshal(byteValue, &data)
+	if err != nil {
+		if utils.ConfigJSON.Verbose > 0 {
+			log.Println(string(byteValue))
+		}
+		log.Printf("Unable to parse JSON Data from AlertManager GET API, error: %v\n", err)
+		return data, err
+	}
+
+	if utils.ConfigJSON.Verbose > 1 {
```

Comment:
This block should go above unmarshal the body since in debug mode we need to see response before we try to unmarshal it. For instance, if response fails to unmarshal this block will never be called.

Reply:
Fixed this.